### PR TITLE
t/t6030: improve test coverage

### PR DESCRIPTION
**Pull request by pranitbauva1997**
Explicitly test whether `git bisect reset` cleans the state properly by
removing every file.

Mentored-by: Lars Schneider <larsxschneider@gmail.com>
Mentored-by: Christian Couder <chriscool@tuxfamily.org>
Signed-off-by: Pranit Bauva <pranit.bauva@gmail.com>

### CODE REVIEW

File: t/t6030-bisect-porcelain.sh [hunk 1]
```diff
@@ -894,4 +894,21 @@ test_expect_success 'bisect start takes options and revs in any order' '
 	test_cmp expected actual
 '
 
+test_expect_success 'git bisect reset cleans its state properly' '
+	git bisect reset &&
+	git bisect start &&
+	git bisect good $HASH1 &&
+	git bisect bad $HASH4 &&
+	! test -d ".git/refs/bisect" &&
```
Comment (chriscool) [inline t/t6030-bisect-porcelain.sh]: No need to test stuff under .git/refs. See your patch that removes .git/refs/bisect for explanations.

**Comment by chriscool** [comment]
Except the comment about stuff under .git/refs, this looks good to me.

File: t/t6030-bisect-porcelain.sh (hunk 1 above)
Comment (pranitbauva1997) [inline t/t6030-bisect-porcelain.sh]: sure!

Comment (chriscool) [inline t/t6030-bisect-porcelain.sh]: If you want, you can use `git for-each-ref` to check that there is no more ref under "refs/bisect/".

Comment (chriscool) [inline t/t6030-bisect-porcelain.sh]: By the way "git-bisect.sh" uses `git for-each-ref`, so you might want to look at how it is used...

Comment (pranitbauva1997) [inline t/t6030-bisect-porcelain.sh]: @chriscool Thanks for pointing it out. I have already used it in my [patch](http://thread.gmane.org/gmane.comp.version-control.git/294520).

Comment (chriscool) [inline t/t6030-bisect-porcelain.sh]: Great! And sorry I hadn't seen the patch.

**Pull request closed without merging**